File: Source/AlphaTab/Rendering/Glyphs/FadeInGlyph.cs
using System;
using AlphaTab.Platform;

namespace AlphaTab.Rendering.Glyphs
{
    class FadeInGlyph : EffectGlyph
    {
        public FadeInGlyph(float x, float y)
            : base(x, y)
        {
        }

        public override void DoLayout()
        {
            base.DoLayout();
            Height = 17 * Scale;
        }

        public override void Paint(float cx, float cy, ICanvas canvas)
        {
            var size = 6 * Scale;
            var width = Math.Max(Width, 14 * Scale);

            var offset = Height / 2;

            canvas.BeginPath();
            canvas.MoveTo(cx + X, cy + Y + offset);
            canvas.QuadraticCurveTo(cx + X + (width / 2), cy + Y + offset, cx + X + width, cy + Y + offset - size);
            canvas.MoveTo(cx + X, cy + Y + offset);
            canvas.QuadraticCurveTo(cx + X + (width / 2), cy + Y + offset, cx + X + width, cy + Y + offset + size);
            canvas.Stroke();
        }
    }
}

File: src/rendering/glyphs/FadeInGlyph.ts
import { ICanvas } from '@src/platform/ICanvas';
import { EffectGlyph } from '@src/rendering/glyphs/EffectGlyph';

export class FadeInGlyph extends EffectGlyph {
    public doLayout(): void {
        super.doLayout();
        this.height = 17 * this.scale;
    }

    public paint(cx: number, cy: number, canvas: ICanvas): void {
        let size: number = 6 * this.scale;
        let width: number = Math.max(this.width, 14 * this.scale);
        let offset: number = this.height / 2;
        canvas.beginPath();
        canvas.moveTo(cx + this.x, cy + this.y + offset);
        canvas.quadraticCurveTo(
            cx + this.x + width / 2,
            cy + this.y + offset,
            cx + this.x + width,
            cy + this.y + offset - size
        );
        canvas.moveTo(cx + this.x, cy + this.y + offset);
        canvas.quadraticCurveTo(
            cx + this.x + width / 2,
            cy + this.y + offset,
            cx + this.x + width,
            cy + this.y + offset + size
        );
        canvas.stroke();
    }
}
